src/app/(profile)/(components)/ProfilesProvider.tsx

```diff
@@ -87,6 +97,53 @@ export const ProfilesProvider = ({ children }: PropsWithChildren) => {
     [filters],
   )
 
+  const refreshProfilesWithProfileViews = (
+    viewerID: string,
+    viewedProfileID: string,
```

Comment:
**issue:** This is complicated and it does a lot of things - I thought that it's a fetch. We should just call this method `handleVisitProfile(profileView: ProfileViewModel)` - profile view will contain info about viewer, and profile, so no need for other params